File: dgx/server-node/src/services/jobStore.js
/**
 * jobStore.js — simple in-memory job store for PDF processing jobs.
 */

'use strict';

class JobStore {
  constructor() {
    /** @type {Map<string, Object>} */
    this._store = new Map();
  }

  create(jobId) {
    const job = {
      jobId,
      status: 'pending', // pending | running | done | error
      resultFile: null,
      error: null,
      createdAt: Date.now(),
    };
    this._store.set(jobId, job);
    return job;
  }

  get(jobId) {
    return this._store.get(jobId) || null;
  }

  update(jobId, updates) {
    const job = this._store.get(jobId);
    if (job) Object.assign(job, updates);
    return job;
  }

  delete(jobId) {
    return this._store.delete(jobId);
  }

  /** Return a snapshot of all jobs (used for TTL cleanup). */
  listAll() {
    return Array.from(this._store.values());
  }
}

const jobStore = new JobStore();
module.exports = { jobStore };

File: dgx/server/services/job_store.py
import threading
import time
from dataclasses import dataclass, field
from typing import Optional


@dataclass
class Job:
    job_id: str
    status: str = "pending"  # pending | running | done | error
    result_file: Optional[str] = None
    error: Optional[str] = None
    created_at: float = field(default_factory=time.time)


class JobStore:
    def __init__(self):
        self._store: dict[str, Job] = {}
        self._lock = threading.Lock()

    def create(self, job_id: str) -> Job:
        job = Job(job_id=job_id)
        with self._lock:
            self._store[job_id] = job
        return job

    def get(self, job_id: str) -> Optional[Job]:
        with self._lock:
            return self._store.get(job_id)

    def update(self, job_id: str, **kwargs) -> Optional[Job]:
        with self._lock:
            job = self._store.get(job_id)
            if job:
                for k, v in kwargs.items():
                    setattr(job, k, v)
        return job

    def delete(self, job_id: str) -> bool:
        with self._lock:
            return self._store.pop(job_id, None) is not None

    def list_all(self) -> list["Job"]:
        """Return a snapshot of all jobs (used for TTL cleanup)."""
        with self._lock:
            return list(self._store.values())


# Module-level singleton
job_store = JobStore()
